packages/cw-paginate/src/lib.rs

```diff
@@ -76,7 +83,38 @@ where
     K: Bounder<'a> + KeyDeserialize<Output = K> + 'static,
     V: serde::de::DeserializeOwned + serde::Serialize,
 {
-    let items = map.keys(deps.storage, start_after.map(Bound::exclusive), None, order);
+    let (range_min, range_max) = match order {
+        Order::Ascending => (start_after.map(Bound::exclusive), None),
+        Order::Descending => (None, start_after.map(Bound::exclusive)),
+    };
+
+    let items = map.keys(deps.storage, range_min, range_max, order);
+    match limit {
+        Some(limit) => Ok(items
+            .take(limit.try_into().unwrap())
+            .collect::<StdResult<_>>()?),
+        None => Ok(items.collect::<StdResult<_>>()?),
+    }
+}
+
+/// Same as `paginate_map` but for use with `SnapshotMap`.
+pub fn paginate_snapshot_map<'a, 'b, K, V, R: 'static>(
```

Comment:
since this deals with a snapshot map, does it make sense to take in an optional height parameter? (not sure if they expose range functions for this, so may be gas inefficient)

Reply:
Good question. Digging into the code a little: snapshot maps seem to only encode the diff. for example "at this height the value was set to X". We could technically reconstruct the state at a given height using this info, but IMO that might be best done as followup work.

Reply:
sounds good to me, we can investigate if the need arises later